sdk/core/Azure.Core/src/Shared/AzureSasCredentialSynchronousPolicy.cs

```diff
@@ -29,13 +29,33 @@ public override void OnSendingRequest(HttpMessage message)
             {
                 signature = signature.Substring(1);
             }
+
             if (!query.Contains(signature))
             {
-                query = string.IsNullOrEmpty(query) ? '?' + signature : query + '&' + signature;
+                // Get the signature history so we know which string to replace in the query if it was previously set.
+                if (message.TryGetProperty(typeof(AzureSasSignatureHistory), out object? previousSignature) &&
+                    previousSignature != null &&
+                    query.Contains((string)previousSignature!))
+                {
+                    query = query.Replace((string)previousSignature!, signature);
+                }
+                else
+                {
+                    // This is the first time we're setting the signature, so we need to add it to the query.
+                    query = string.IsNullOrEmpty(query) ? '?' + signature : query + '&' + signature;
+
+                    // Store the signature in the message property bag so we can replace it if it changes.
+                    message.SetProperty(typeof(AzureSasSignatureHistory), signature);
+                }
                 message.Request.Uri.Query = query;
-            }
 
-            base.OnSendingRequest(message);
+                base.OnSendingRequest(message);
+            }
         }
+
+        /// <summary>
+        /// A marker class used to represent AzureSasCredential signature history on the <see cref="HttpMessage"/> property bag.
+        /// </summary>
+        internal class AzureSasSignatureHistory { }
```

Comment:
Does it need to be internal?  One of the goals of this design, I think, was that only the policy can get and set these values.  If the type is private, it is easier to achieve that.